cmd/gpud/up/command.go

```diff
@@ -30,14 +31,33 @@ func Command(cliContext *cli.Context) (retErr error) {
 		return err
 	}
 
+	// step 1.
+	// perform "login" if and only if configured
 	if cliContext.String("token") != "" {
 		log.Logger.Debugw("non-empty --token provided, logging in")
 		if lerr := cmdlogin.Command(cliContext); lerr != nil {
 			return lerr
 		}
 		log.Logger.Debugw("successfully logged in")
+	} else {
+		log.Logger.Infow("no --token provided, skipping login")
 	}
 
+	// step 2.
+	// perform "join" if and only if configured
+	nodeGroup := cliContext.String("node-group")
+	if nodeGroup != "" {
+		log.Logger.Debugw("non-empty --node-group provided, joining")
+		if jerr := cmdjoin.Command(cliContext); jerr != nil {
+			return jerr
+		}
+		log.Logger.Debugw("successfully joined")
+	} else {
+		log.Logger.Infow("no --token provided, skipping join")
+	}
```

Comment:
if the nodegroup is specified, we should pass it in login request, so that we only need one request to login + join.